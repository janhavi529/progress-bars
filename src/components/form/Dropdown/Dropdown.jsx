import React from 'react';
import PropTypes from 'prop-types';

const dropdown = ({ label: dropdownLabel, value, options, onChange }) => (
    <div className="form-group mx-2 text-center">
        <label htmlFor="selectMenu">{dropdownLabel}</label>
        <select id="selectMenu"
            className="form-control"
            value={value}
            onChange={onChange}>
            {
                options.map(({id, label: optionLabel}) => (
                    <option key={id} value={id}>{optionLabel}</option>
                ))
            }
        </select>
    </div>
);

dropdown.propTypes = {
    label: PropTypes.string,
    value: PropTypes.number,
    options: PropTypes.arrayOf(PropTypes.shape({
        id: PropTypes.number,
        label: PropTypes.string,
    })),
    onChange: PropTypes.func,
};

dropdown.defaultProps = {
    value: 1,
};

export default dropdown;