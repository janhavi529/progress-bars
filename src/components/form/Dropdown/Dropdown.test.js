import React from 'react';
import { shallow } from 'enzyme';
import Dropdown from './Dropdown';

describe('Dropdown', () => {
    const props = {
        label: 'Select Progress Bar',
        value: 1,
        options: [
            {
                id: 1,
                label: 'Progress Bar #1'
            }, 
            {
                id: 2,
                label: 'Progress Bar #2'
            }, 
            {
                id: 3,
                label: 'Progress Bar #3'
            }],
        onChange: () => {}
    };

    it('Renders dropdown with default props', () => {
        const wrapper = shallow(<Dropdown {...props}/>);
        expect(wrapper).toMatchSnapshot();
    });
});