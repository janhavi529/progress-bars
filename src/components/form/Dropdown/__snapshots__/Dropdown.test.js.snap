// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown Renders dropdown with default props 1`] = `ShallowWrapper {}`;
